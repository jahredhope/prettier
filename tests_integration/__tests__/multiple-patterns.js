"use strict";

const runPrettier = require("../runPrettier");

test("multiple patterns", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "directory/**/*.js",
    "other-directory/**/*.js",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("multiple patterns with non exists pattern", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "directory/**/*.js",
    "non-existent.js",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("multiple patterns with ignore nested directories pattern", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "**/*.js",
    "!**/nested-directory/**",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("multiple patterns by with ignore pattern, ignores node_modules by default", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "**/*.js",
    "!directory/**",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("multiple patterns by with ignore pattern, ignores node_modules by with ./**/*.js", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "./**/*.js",
    "!./directory/**",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("multiple patterns by with ignore pattern, doesn't ignore node_modules with --with-node-modules flag", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "**/*.js",
    "!directory/**",
    "-l",
    "--with-node-modules"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("no errors on empty patterns", () => {
  const result = runPrettier("cli/multiple-patterns");

  expect(result.status).toEqual(0);
});

test("multiple patterns, throw error and exit with non zero code on non existing files", () => {
  const result = runPrettier("cli/multiple-patterns", [
    "non-existent.js",
    "other-non-existent.js",
    "-l"
  ]);

  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(2);
});
