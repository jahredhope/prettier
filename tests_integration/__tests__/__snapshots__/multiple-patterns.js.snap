// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple patterns 1`] = `
"directory/file.js
directory/nested-directory/nested-directory-file.js
other-directory/file.js
other-directory/nested-directory/nested-directory-file.js
"
`;

exports[`multiple patterns by with ignore pattern, doesn't ignore node_modules with --with-node-modules flag 1`] = `
"node_modules/node-module.js
other-directory/file.js
other-directory/nested-directory/nested-directory-file.js
other-regular-modules.js
regular-module.js
"
`;

exports[`multiple patterns by with ignore pattern, ignores node_modules by default 1`] = `
"other-directory/file.js
other-directory/nested-directory/nested-directory-file.js
other-regular-modules.js
regular-module.js
"
`;

exports[`multiple patterns by with ignore pattern, ignores node_modules by with ./**/*.js 1`] = `
"./other-directory/file.js
./other-directory/nested-directory/nested-directory-file.js
./other-regular-modules.js
./regular-module.js
"
`;

exports[`multiple patterns with ignore nested directories pattern 1`] = `
"directory/file.js
other-directory/file.js
other-regular-modules.js
regular-module.js
"
`;

exports[`multiple patterns with non exists pattern 1`] = `
"directory/file.js
directory/nested-directory/nested-directory-file.js
"
`;

exports[`multiple patterns, throw error and exit with non zero code on non existing files 1`] = `
"No matching files. Patterns tried: non-existent.js other-non-existent.js !**/node_modules/** !./node_modules/**
"
`;
